### ISSUE TITLE
Whitespace Between CJK & Latin not Working with Some Raw Text

### ISSUE THREAD

**Issue by jiaojiaodubai** [typst/typst#2702]
### Description

Minimum working example

```typst
汉字`(code)`汉字

汉字 `(code)` 汉字
```

I tried to write the name of a function in `raw text` with parentheses, and then I found that there were no spaces between it and the subsequent CJK text. Logically speaking, regardless of the content in `raw text`, it should be considered as Latin text.

### Operating system

Windows

### Typst version

- [X] I am using the latest version of Typst

**Comment by Enter-tainer** [comment]
currently it only considers texts within the text element.

I dont know if this is possible. Maybe we should also consider math equation and so on.

**Comment by Enter-tainer** [comment]
> Logically speaking, regardless of the content in raw text, it should be considered as Latin text.

Do you mean we should always add space between code blocks and cjk character?

**Comment by jiaojiaodubai** [comment]
Yes, as mentioned in [the reference document](https://typst.app/docs/reference/text/raw/), the text in the raw text will be in mono and is typically used to display code. Of course, CJK text can also be placed in raw text and displayed as mono, but this is an edge case. If someone really wants to display CJK text in mono, it should use methods like `#set text` instead of relying on raw text.

**Comment by Enter-tainer** [comment]
maybe this issue should be renamed to cjk-raw spacing or something similar
